service/matching/taskListManager.go

```diff
@@ -204,6 +205,8 @@ func (c *taskListManagerImpl) persistAckLevel() error {
 		},
 	}
 	c.Unlock()
+	c.persistenceLock.Lock()
```

Comment:
also this looks like a cassandra specific limitation (atleast for now). How do you feel about moving this lock to the persistence layer ?